### PR TITLE
updated environment, nodes to use node affinity scheduling

### PR DESCRIPTION
**Pull request by scottyhq**

### CODE REVIEW

**Comment by scottyhq** [comment]
building was failing with some network issues: 

```
CondaHTTPError: HTTP 504 GATEWAY_TIMEOUT for url <https://conda.anaconda.org/conda-forge/label/broken/linux-64/notebook-5.7.4-py37_1000.tar.bz2>
Elapsed: 00:59.798074
CF-RAY: 4fc2a1511e043832-ATL

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
```

File: deployments/nasa/config/common.yaml [hunk 1]
```diff
@@ -1,5 +1,12 @@
 pangeo:
   jupyterhub:
+    scheduling:
+      userPods:
+        nodeAffinity:
+          matchNodePurpose: require
+      corePods:
+        nodeAffinity:
+          matchNodePurpose: require
```
Comment (jhamman) [inline deployments/nasa/config/common.yaml:9]: Nice! What labels / taints do you have on your nodes?

Comment (scottyhq) [reply deployments/nasa/config/common.yaml:9]:
```
    labels:
      node-role.kubernetes.io/role: hub
      hub.jupyter.org/node-purpose: core

    labels:
      node-role.kubernetes.io/role: user-notebook
      hub.jupyter.org/node-purpose: user
    taints:
      hub.jupyter.org/dedicated: 'user:NoSchedule'

    labels:
      node-role.kubernetes.io/role: dask-worker
      k8s.dask.org/node-purpose: worker
    taints:
      k8s.dask.org/dedicated: 'worker:NoSchedule'

```

Comment (jhamman) [reply deployments/nasa/config/common.yaml:9]: whoohoo!

**Pull request merged**